src/lib/loader.js

```diff
@@ -99,13 +108,27 @@ function forceFocus(el) {
 /**
  * Replaces the current #content element with new partial content.
  *
- * @param {!Object} partial
+ * @param {{offline: boolean, html: string}} partial
  */
 function updateDom(partial) {
+  const incomingDocument = new DOMParser().parseFromString(
+    partial.html,
+    'text/html',
+  );
+
+  const incomingResourceVersion = incomingDocument.getAttribute('data-version');
```

Comment:
I don't understand what the resource version does

Reply:
ok I now understand BUT it might be worth adding a big comment here cuz it's important.